brave-core/src/main/java/com/github/kristofa/brave/internal/Propagation.java

```diff
@@ -0,0 +1,61 @@
+package com.github.kristofa.brave.internal;
+
+import com.github.kristofa.brave.SpanId;
+import com.github.kristofa.brave.TraceData;
+
+/**
+ * Injects and extracts {@link SpanId span identifiers} sent in-band across process boundaries.
+ *
+ * <p>Implementations usually wrap Http, Messaging or RPC headers.
+ *
+ * @param <K> Usually, but not always a String
+ */
+public interface Propagation<K> {
+
+  interface Factory {
```

Comment:
while possible to change propagation from B3 to something else, I've left customizing this out of it for now. Mainly I want to finish the other change about specifying keys. Once that's in, it will be more clear which types need to hold this responsibility. Until then, it is status quo (B3 only)